partiql-eval/src/main/kotlin/org/partiql/eval/internal/helpers/PErrors.kt

```diff
@@ -0,0 +1,226 @@
+package org.partiql.eval.internal.helpers
+
+import org.partiql.spi.errors.PError
+import org.partiql.spi.errors.PErrorException
+import org.partiql.spi.errors.PErrorKind
+import org.partiql.spi.errors.Severity
+import org.partiql.spi.function.Function
+import org.partiql.spi.types.PType
+
+internal object PErrors {
+
+    /**
+     * Returns a PErrorException with code: [PError.NUMERIC_VALUE_OUT_OF_RANGE].
+     */
+    fun numericOutOfRangeException(value: String, type: PType): PErrorException {
+        val pError = numericOutOfRange(value, type)
+        return PErrorException(pError)
+    }
```

Comment:
wonder if we'll ever make these constructor functions public within SPI? there is a lot of overlap between the versions

Reply:
The intention is to hide it for now unless a customer explicitly wants it. Trying to keep our public API minimal.

Reply:
Agree w/ keeping the public API minimal. We can always add it in the future. I guess we'll see how the usages evolve over time as we add more helpers.